index.bs

```diff
@@ -232,27 +232,130 @@ Non-Functional Requirements {#requirements-non-functional}
+
+To learn further metadata, an endpoint may send a endpoint-info-request
+message (see Appendix A) and receive back a endpoint-info-response message.  The
+messages may contain the following information with the following meaning:
+
+- display-name: The display name of the responding endpoint intended to be
+  displayed to a user by the requesting endpoint.  If the responding endpoint is
+  not yet authenticated, the requesting endpoint should make UI affordance for
+  indicating to the user that the display name is not yet authenticated.  If the
+  responding endpoint changes its display name, the requesting endpoint should
+  make UI affordance for indicating to the user that the display name has
+  changed. 
+
+- model-name: The model name of the device (if it is a device), used mainly for
+  debugging purposes, but may be displayed to the user of the requesting
+  endpoint.
+
+<!-- TODO: Add device type and/or capabilities -->
+
+If a discovering endpoint (acting as a QUIC client) wishes to receive messages
+from an advertising endpoint (acting as a QUIC server) or an advertising
+endpoint (acting as a QUIC server) wishes to send message to a discovering
+endpoint (acting as a client endpoint), it may wish to keep the QUIC connection
+alive.  Once neither side needs to keep the connection alive for the purposes of
+sending or receiving messages, the connection should be closed.  In order to
+keep a QUIC connection alive, an endpoint may send an endpoint-status-request
+message, and any endpoint that receives an endpoint-status-request message
+should send an endpoint-status-response message.  Such messages should be sent
+every 25 seconds.  If, however, other mechanisms or traffic (such as other
+requests and responses) are serving the purpose of keeping the connection alive,
+these messages should be delayed or not used at all.  A QUIC connection should
+not be closed because of idle timeout unless no such mechanisms or traffic has
+been received from the remote side for 35 seconds (allowing for a keep-alive
+frequency of 25 seconds and message delivery delay of 10 seconds).
+
+If a client endpoint wishes to send messages to a server endpoint, the client
+endpoint can connect to the server endpoint "on demand": it does not need to
+keep the connection alive.
+
+The endpoint-info-response message and endpoint-status-response messages may be
+extended to include additional information not defined in this spec.  If done
+adhoc by applications and not in future specs, keys should be chosen to avoid
+collision, such as by choosing large integers or long strings.  
+
+Messages delivery using CBOR and QUIC streams {#control}
+========================================================
+
+Messages are serialized using [CBOR](https://tools.ietf.org/html/rfc7049).  To
+send a group of messages in order, that group of messages must be sent in one
+QUIC stream.  Different groups of messages which are ordered within the group
```

Comment:
This sentence didn't really make sense to me, possibly because of the double meaning of "group".  Can you give an example or clarify further?

Reply:
I updated it to make it more clear.